src/plugins/lua/luaplugin.h

```diff
@@ -79,9 +70,66 @@ class LUASHARED_EXPORT LuaPlugin : public Tiled::WritableMapFormat
     void writePolygon(LuaTableWriter &, const Tiled::MapObject *);
     void writeTextProperties(LuaTableWriter &, const Tiled::MapObject *);
 
-    QString mError;
-    QDir mMapDir;     // The directory in which the map is being saved
+protected:
+    QDir mMapDir;
     Tiled::GidMapper mGidMapper;
+
 };
 
+class LUASHARED_EXPORT LuaPlugin : public Tiled::Plugin
+{
+    Q_OBJECT
+    Q_INTERFACES(Tiled::Plugin)
+    Q_PLUGIN_METADATA(IID "org.mapeditor.Plugin" FILE "plugin.json")
+
+public:
+    void initialize() override;
+};
+
+
+class LUASHARED_EXPORT LuaMapFormat : public Tiled::WritableMapFormat, public LuaUtilWriter
+{
+    Q_OBJECT
+//    Q_INTERFACES(Tiled::MapFormat)
+
+public:
+    LuaMapFormat(QObject *parent = nullptr) : Tiled::WritableMapFormat(parent) {}
+
+    bool supportsFile(const QString &fileName) const override;
+
+    bool writeMap(const Tiled::Map *map, const QString &fileName) override;
+
+    QString nameFilter() const override;
+    QString shortName() const override;
+    QString errorString() const override;
+
+protected:
+    QString mError;
+
+};
+
+
+class LUASHARED_EXPORT LuaTilesetFormat : public Tiled::WritableTilesetFormat, public LuaUtilWriter
+{
+    Q_OBJECT
+//    Q_INTERFACES(Tiled::WritableTilesetFormat)
```

Comment:
You can remove this. There is no such interface declared, and it isn't useful on top of the `Tiled::TilesetFormat` interface.

Reply:
Done